#!/bin/sh /usr/share/dpatch/dpatch-run
## 11_fix-misc-gcc-warnings.dpatch by Henrique de Moraes Holschuh <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix bad C code that causes warnings

@DPATCH@
diff -urNad --exclude=CVS --exclude=.svn ./pcard/pcardext/pcardext.c ./pcard/pcardext/pcardext.c
--- ./pcard/pcardext/pcardext.c	2005-09-17 13:51:49.000000000 -0300
+++ ./pcard/pcardext/pcardext.c	2005-09-17 13:54:16.731760274 -0300
@@ -20,7 +20,7 @@
 Requires:
 Python 2.2+
 
-/*****************************************************************************/
+\*****************************************************************************/
 
 #include <Python.h>
 #include <structmember.h>
diff -urNad --exclude=CVS --exclude=.svn ./scan/sane/pml.c ./scan/sane/pml.c
--- ./scan/sane/pml.c	2005-09-17 13:54:12.000000000 -0300
+++ ./scan/sane/pml.c	2005-09-17 13:54:41.978587853 -0300
@@ -1057,7 +1057,8 @@
    else if ((hpaio->page_done || hpaio->pml_done) && !hpaio->ip_done)
    {
       /* No more scan data, flush ipconvert pipeline. */
-      input = 0;
+      input = NULL;
+      inputAvail = 0;
       wResult = ipConvert(hpaio->hJob, inputAvail, input, &inputUsed, &inputNextPos, outputAvail, output, &outputUsed, &outputThisPos);
       if (wResult & (IP_INPUT_ERROR | IP_FATAL_ERROR))
       {
